tests/Feature/Api/PublicScheduleTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use App\Models\Lecturing;
+use Carbon\Carbon;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class PublicScheduleTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function can_get_schedules_within_specified_date_range() 
+    {
+        $lecturing = factory(Lecturing::class)->create([
+            'audience_code' => Lecturing::AUDIENCE_FRIDAY,
+            'date' => Carbon::tomorrow()->format('Y-m-d'),
+        ]);
+
+        $unlistedLecturing = factory(Lecturing::class)->create([
+            'audience_code' => Lecturing::AUDIENCE_PUBLIC,
+            'date' => Carbon::yesterday()->format('Y-m-d'),
+        ]);
+
+        $startDate = Carbon::now()->format('Y-m-d');
+        $endDate = Carbon::now()->addDays(7)->format('Y-m-d');
+
+        $this->getJson(route('api.schedules.index', [
+            'startDate' => $startDate,
+            'endDate' => $endDate,
```

Comment:
Ini testingnya error, bagian yang ini perlu kita sesuaikan pakai snake_case mas.
```suggestion
            'start_date' => $startDate,
            'end_date' => $endDate,
```